### PR TITLE
PP-864 Allow the ability to search charges by email 

### PR DESCRIPTION
**Pull request by PauloPortugal**
## WHAT
_A brief description of the pull request:_

* pay-connector is already returning the `email` field, so we just capture that field and pass it accross to the client.    
* The email will do a partial and full match.
* Amending the unit tests for search charges by email

## HOW 
_Steps to test or reproduce:_
```
cd $WORKSPACE
msl reset && msl -a up && ./run-endtoend.sh

### CODE REVIEW

File: src/main/java/uk/gov/pay/api/model/PaymentForSearchResult.java [hunk 1]
```diff
@@ -12,9 +12,9 @@
     private PaymentLinksForSearch links = new PaymentLinksForSearch();
 
     public PaymentForSearchResult(String chargeId, long amount, PaymentState state, String returnUrl, String description,
-                                  String reference, String paymentProvider, String createdDate,
+                                  String reference, String email, String paymentProvider, String createdDate,
```
Comment (rahul-vinayak) [inline src/main/java/uk/gov/pay/api/model/PaymentForSearchResult.java]: This constructor's arguments are ever growing now, 14 in total, we need to think of an alternate at some point (builder?), but may be for a later refactoring task. Fine as such.

File: src/main/java/uk/gov/pay/api/model/PaymentWithAllLinks.java [hunk 2]
```diff
@@ -13,10 +13,10 @@
     private PaymentLinks links = new PaymentLinks();
 
     private PaymentWithAllLinks(String chargeId, long amount, PaymentState state, String returnUrl, String description,
-                                String reference, String paymentProvider, String createdDate, RefundSummary refundSummary,
-                                List<PaymentConnectorResponseLink> paymentConnectorResponseLinks,
+                                String reference, String email, String paymentProvider, String createdDate,
```
Comment (rahul-vinayak) [inline src/main/java/uk/gov/pay/api/model/PaymentWithAllLinks.java]: Minor: This one has grown to 15 arguments as well, a refactoring task later perhaps

File: src/test/java/uk/gov/pay/api/validation/PaymentSearchValidatorTest.java [hunk 3]
```diff
@@ -11,64 +11,72 @@
 
     @Rule
     public ExpectedException expectedException = ExpectedException.none();
+    private static final String SUCCESSFUL_TEST_EMAIL = "alice.111@mail.fake";
+    private static final String INSUCCESSFUL_TEST_EMAIL = randomAlphanumeric(255) + "@mail.fake";
```
Comment (rahul-vinayak) [inline src/test/java/uk/gov/pay/api/validation/PaymentSearchValidatorTest.java]: pedantic: s/IN/UN  
:-D

Comment (PauloPortugal) [inline src/test/java/uk/gov/pay/api/validation/PaymentSearchValidatorTest.java]: Thanks for noticing the spelling mistake. Corrected

File: src/main/java/uk/gov/pay/api/model/PaymentWithAllLinks.java (hunk 2 above)
Comment (PauloPortugal) [inline src/main/java/uk/gov/pay/api/model/PaymentWithAllLinks.java]: As discussed this needs to be refactored. I will probably raise another PR for this matter, in the meantime I want to have this story delivered.

**Pull request merged**